x/signal/keeper_test.go

```diff
@@ -291,32 +313,118 @@ func TestThresholdVotingPower(t *testing.T) {
+		require.NoError(t, err)
+		_, err = upgradeKeeper.SignalVersion(ctx, &types.MsgSignalVersion{ValidatorAddress: testutil.ValAddrs[3].String(), Version: 2})
+		require.NoError(t, err)
+
+		// This TryUpgrade should succeed.
+		_, err = upgradeKeeper.TryUpgrade(goCtx, &types.MsgTryUpgrade{})
+		require.NoError(t, err)
+
+		// This TryUpgrade should fail because an upgrade is pending.
+		_, err = upgradeKeeper.TryUpgrade(goCtx, &types.MsgTryUpgrade{})
+		require.Error(t, err)
+		require.ErrorIs(t, err, types.ErrUpgradePending)
+	})
+
+	t.Run("should return an error if quorum version is less than or equal to the current version", func(t *testing.T) {
+		upgradeKeeper, ctx, _ := setup(t)
+		goCtx := sdk.WrapSDKContext(ctx)
+
+		_, err := upgradeKeeper.SignalVersion(ctx, &types.MsgSignalVersion{ValidatorAddress: testutil.ValAddrs[0].String(), Version: 1})
+		require.NoError(t, err)
+		_, err = upgradeKeeper.SignalVersion(ctx, &types.MsgSignalVersion{ValidatorAddress: testutil.ValAddrs[1].String(), Version: 1})
+		require.NoError(t, err)
+		_, err = upgradeKeeper.SignalVersion(ctx, &types.MsgSignalVersion{ValidatorAddress: testutil.ValAddrs[2].String(), Version: 1})
+		require.NoError(t, err)
+		_, err = upgradeKeeper.SignalVersion(ctx, &types.MsgSignalVersion{ValidatorAddress: testutil.ValAddrs[3].String(), Version: 1})
+		require.NoError(t, err)
+
+		_, err = upgradeKeeper.TryUpgrade(goCtx, &types.MsgTryUpgrade{})
+		require.Error(t, err)
+		require.ErrorIs(t, err, types.ErrInvalidUpgradeVersion)
+	})
+}
+
+func TestGetUpgrade(t *testing.T) {
+	upgradeKeeper, ctx, _ := setup(t)
+	goCtx := sdk.WrapSDKContext(ctx)
+
+	t.Run("should return an empty upgrade if no upgrade is pending", func(t *testing.T) {
+		got, err := upgradeKeeper.GetUpgrade(ctx, &types.QueryGetUpgradeRequest{})
+		require.NoError(t, err)
+		assert.Nil(t, got.Upgrade)
+	})
+
+	t.Run("should return an upgrade if an upgrade is pending", func(t *testing.T) {
+		_, err := upgradeKeeper.SignalVersion(ctx, &types.MsgSignalVersion{ValidatorAddress: testutil.ValAddrs[0].String(), Version: 2})
+		require.NoError(t, err)
+		_, err = upgradeKeeper.SignalVersion(ctx, &types.MsgSignalVersion{ValidatorAddress: testutil.ValAddrs[1].String(), Version: 2})
+		require.NoError(t, err)
+		_, err = upgradeKeeper.SignalVersion(ctx, &types.MsgSignalVersion{ValidatorAddress: testutil.ValAddrs[2].String(), Version: 2})
+		require.NoError(t, err)
+		_, err = upgradeKeeper.SignalVersion(ctx, &types.MsgSignalVersion{ValidatorAddress: testutil.ValAddrs[3].String(), Version: 2})
+		require.NoError(t, err)
+
+		// This TryUpgrade should succeed.
+		_, err = upgradeKeeper.TryUpgrade(goCtx, &types.MsgTryUpgrade{})
+		require.NoError(t, err)
+
+		got, err := upgradeKeeper.GetUpgrade(ctx, &types.QueryGetUpgradeRequest{})
+		require.NoError(t, err)
+		assert.Equal(t, uint64(2), got.Upgrade.AppVersion)
```

Comment:
[nit]
If there is a constant for the app version e.g., [this](https://github.com/celestiaorg/celestia-app/blob/4c732d312032d96466b06e5f5b7db98268f2b1b1/pkg/appconsts/v2/app_consts.go#L4), I'd suggest using the const instead of literals.

Reply:
8e9c37d86a8c5efd3522f4298d3802c124e3c4a2